x-pack/plugins/enterprise_search/server/routes/app_search/search_settings.test.ts

```diff
@@ -0,0 +1,216 @@
+      path: '/api/app_search/engines/{engineName}/search_settings',
+      payload: 'body',
+    });
+
+    beforeEach(() => {
+      registerSearchSettingsRoutes({
+        ...mockDependencies,
+        router: mockRouter.router,
+      });
+    });
+
+    it('creates a request to enterprise search', () => {
+      mockRouter.callRoute({
+        params: { engineName: 'some-engine' },
+        body: searchSettings,
+      });
+
+      expect(mockRequestHandler.createRequest).toHaveBeenCalledWith({
+        path: '/as/engines/:engineName/search_settings',
+      });
+    });
+
+    describe('validates', () => {
+      it('correctly', () => {
+        const request = { body: searchSettings };
+        mockRouter.shouldValidate(request);
+      });
+
+      it('missing required fields', () => {
+        const request = { body: {} };
+        mockRouter.shouldThrow(request);
+      });
+    });
+  });
+
+  describe('POST /api/app_search/engines/{name}/search_settings/reset', () => {
+    const mockRouter = new MockRouter({
+      method: 'post',
+      path: '/api/app_search/engines/{engineName}/search_settings/reset',
+    });
+
+    beforeEach(() => {
+      registerSearchSettingsRoutes({
+        ...mockDependencies,
+        router: mockRouter.router,
+      });
+    });
+
+    it('creates a request to enterprise search', () => {
+      mockRouter.callRoute({
+        params: { engineName: 'some-engine' },
+      });
+
+      expect(mockRequestHandler.createRequest).toHaveBeenCalledWith({
+        path: '/as/engines/:engineName/search_settings/reset',
+      });
+    });
+  });
+
+  describe('POST /api/app_search/engines/{name}/search_settings_search', () => {
+    const mockRouter = new MockRouter({
+      method: 'post',
+      path: '/api/app_search/engines/{engineName}/search_settings_search',
+      payload: 'body',
+    });
+
+    beforeEach(() => {
+      registerSearchSettingsRoutes({
+        ...mockDependencies,
+        router: mockRouter.router,
+      });
+    });
+
+    it('creates a request to enterprise search', () => {
+      mockRouter.callRoute({
+        params: { engineName: 'some-engine' },
+        body: searchSettings,
+      });
+
+      expect(mockRequestHandler.createRequest).toHaveBeenCalledWith({
+        path: '/as/engines/:engineName/search_settings_search',
+      });
+    });
+
+    describe('validates body', () => {
+      it('correctly', () => {
+        const request = {
+          body: {
+            boosts,
+            search_fields: searchFields,
+          },
+        };
+        mockRouter.shouldValidate(request);
+      });
+
+      it('missing required fields', () => {
+        const request = { body: {} };
+        mockRouter.shouldThrow(request);
+      });
+    });
+
+    describe('validates query', () => {
+      const queryRouter = new MockRouter({
+        method: 'post',
+        path: '/api/app_search/engines/{engineName}/search_settings_search',
+        payload: 'query',
+      });
```

Comment:
Hmm I think I need to rethink how I handle payload validation if we're having to create a brand new router for validating body vs query etc. It hadn't occurred to past-me that we'd have multiple types of payloads to validate. 🤦‍♀️ 

Basically what I'm thinking of changing API-wise is removing the `payload` from MockRouter instantiation and only passing it in the actual `shouldValidate`/`shouldThrow` fn. That way we'll only need to create a single mock router per route.

I'll follow up with this PR with that refactor if that sounds good to you!

Reply:
Yeah I think that sounds great Constance, ty!